### PR TITLE
[16] Add exponential backoff for api.take

### PR DESCRIPTION
**Pull request by Alex-Kuz**
- add parameter 'wait_factor' to option table
- it stops trying if the next attempt is after timeout

### CODE REVIEW

File: sharded_queue/api.lua [hunk 1]
```diff
@@ -92,6 +92,12 @@ function sharded_tube.take(self, timeout, options)
     end
     options.tube_name = self.tube_name
 
+    local wait_factor = 1
```
Comment (sharonovd) [inline sharded_queue/api.lua:95]: Default value should not be 1, or it won't `fix` anything by default

File: sharded_queue/api.lua [hunk 2]
```diff
@@ -92,6 +92,12 @@ function sharded_tube.take(self, timeout, options)
...
+    if options.wait_factor and options.wait_factor > 1 then
```
Comment (sharonovd) [inline sharded_queue/api.lua:96]: Clients should not bother passing wait_factor with each request

File: sharded_queue/api.lua [hunk 3]
```diff
@@ -121,11 +127,16 @@ function sharded_tube.take(self, timeout, options)
 
         if task ~= nil then return task end
 
-        fiber.sleep(wait_part)
+        if take_timeout < wait_part * 1e6 then
```
[the remark this answers is not in the document]
Comment (Alex-Kuz) [reply sharded_queue/api.lua:130]: This is necessary to abort attempts if the next call is scheduled after the timeout.

File: sharded_queue/api.lua (hunk 1 above)
Comment (Alex-Kuz) [reply sharded_queue/api.lua]: Default value is 2.

**Pull request merged**